### PR TITLE
Fivetran Integration

### PR DESCRIPTION
**Pull request by mbliakharskyi**
## Description

Fivetran Integration

1. This integration includes the implementation of authentication flow between the Elba and the Fivetran and the corresponding tests.
2. Function required to fetch all the users of the Fivetran tool and the complete test suite related to it.
3. Implementation of Inngest functions for syncing users, and scheduling sync users.
4. Implementation of rate-limit middleware.
5. Implementation of webhook to delete user
All the tests related to specific functions are included.

## Additional Notes

N/A

## Checklist:

Before you create this PR, confirm all the requirements listed below by checking the checkboxes like this (`[x]`).

- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests to cover the new feature or fixes.

### CODE REVIEW

File: apps/fivetran/src/inngest/middlewares/rate-limit-middleware.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { InngestMiddleware, RetryAfterError } from 'inngest';
```
Comment (athielba) [inline apps/fivetran/src/inngest/middlewares/rate-limit-middleware.ts:1]: it is missing the test

File: apps/fivetran/src/inngest/middlewares/rate-limit-middleware.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+          transformOutput: (ctx) => {
+            const {
+              result: { error, ...result },
+              ...context
+            } = ctx;
+
+            // Check if the error is a rate limit error (HTTP Status 429)
+            if (error instanceof FivetranError && error.response?.status === 429) {
+              // This is a simplified approach; adjust logic based on actual rate limit policy details
+              const retryAfter = error.response.headers['x-ratelimit-remaining'] as string; // Retry after the interval if no remaining calls
```
Comment (athielba) [inline apps/fivetran/src/inngest/middlewares/rate-limit-middleware.ts:19]: I don't understand how it is equivalent  to `retryAfter`, based on this [doc](https://fivetran.com/docs/rest-api/rate-limiting), it provides `Retry-After` HTTP header, therefore you could use it, even the header the you used is wrong, it is `X-Rate-Limit-Remaining`

File: apps/fivetran/src/inngest/functions/users/delete-users.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import { eq } from 'drizzle-orm';
+import { NonRetriableError } from 'inngest';
+import { db } from '@/database/client';
+import { Organisation } from '@/database/schema';
+import { inngest } from '@/inngest/client';
+import { deleteUser } from '@/connectors/users';
+
+export const deleteSourceUsers = inngest.createFunction(
```
[the remark this answers is not in the document]
Comment (mbliakharskyi) [reply apps/fivetran/src/inngest/functions/users/delete-users.ts:8]: Okay

File: apps/fivetran/src/inngest/functions/users/delete-users.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  {
+    id: 'delete-users',
```
[the remark this answers is not in the document]
Comment (mbliakharskyi) [reply apps/fivetran/src/inngest/functions/users/delete-users.ts:10]: Got it!

File: apps/fivetran/src/inngest/functions/users/schedule-users-synchronize.test.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
+import { expect, test, describe, beforeAll, vi, afterAll } from 'vitest';
```
[the remark this answers is not in the document]
Comment (mbliakharskyi) [reply apps/fivetran/src/inngest/functions/users/schedule-users-synchronize.test.ts:1]: done

File: apps/fivetran/src/inngest/functions/users/schedule-users-synchronize.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
+import { env } from '@/env';
+import { db } from '@/database/client';
+import { Organisation } from '@/database/schema';
+import { inngest } from '../../client';
+
+export const scheduleUsersSynchronize = inngest.createFunction(
+  { id: 'schedule-users-syncs' },
```
[the remark this answers is not in the document]
Comment (mbliakharskyi) [reply apps/fivetran/src/inngest/functions/users/schedule-users-synchronize.ts:7]: done

**Pull request merged**